stl/inc/span

```diff
@@ -563,62 +572,64 @@ public:
         }
 #if _CONTAINER_DEBUG_LEVEL > 0
         else {
-            _STL_VERIFY(_Offset <= this->size(), "Offset out of range in span::subspan()");
+            _STL_VERIFY(_Offset <= size(), "Offset out of range in span::subspan()");
 
             if constexpr (_Count != dynamic_extent) {
-                _STL_VERIFY(_Count <= this->size() - _Offset, "Count out of range in span::subspan()");
+                _STL_VERIFY(_Count <= size() - _Offset, "Count out of range in span::subspan()");
             }
         }
 #endif // _CONTAINER_DEBUG_LEVEL > 0
         using _ReturnType = span<element_type,
             _Count != dynamic_extent ? _Count : (_Extent != dynamic_extent ? _Extent - _Offset : dynamic_extent)>;
-        return _ReturnType{_Mydata + _Offset, _Count == dynamic_extent ? this->size() - _Offset : _Count};
+        return _ReturnType{_Mydata + _Offset, _Count == dynamic_extent ? size() - _Offset : _Count};
     }
 
     _NODISCARD constexpr auto subspan(const size_type _Offset, const size_type _Count = dynamic_extent) const noexcept
     /* strengthened */ {
 #if _CONTAINER_DEBUG_LEVEL > 0
-        _STL_VERIFY(_Offset <= this->size(), "Offset out of range in span::subspan(offset, count)");
-        _STL_VERIFY(_Count == dynamic_extent || _Count <= this->size() - _Offset,
+        _STL_VERIFY(_Offset <= size(), "Offset out of range in span::subspan(offset, count)");
+        _STL_VERIFY(_Count == dynamic_extent || _Count <= size() - _Offset,
             "Count out of range in span::subspan(offset, count)");
 #endif // _CONTAINER_DEBUG_LEVEL > 0
         using _ReturnType = span<element_type, dynamic_extent>;
-        return _ReturnType{_Mydata + _Offset, _Count == dynamic_extent ? this->size() - _Offset : _Count};
+        return _ReturnType{_Mydata + _Offset, _Count == dynamic_extent ? size() - _Offset : _Count};
     }
 
     // [span.obs] Observers
+    using _Span_extent_type<_Ty, _Extent>::size;

```

Comment:
Perhaps it's worth moving the `private` section up, so you can avoid repetition and say `_Mybase` here. (For consistency, I dislike seeing two different ways of referring to a base class in a class definition.)

Reply:
That was what I tried and it didnt work  for some reason

Reply:
Ah the reason is we want to access the function publicly because this is essentially `span::size()`

Reply:
Trying to say `_Mybase` doesn't work without moving the typedef up, because the class definition is still parsed top-down. (It's confusing because member function bodies can refer to things that haven't been declared yet, so we often think of anything in a class definition being able to refer to anything else, but that's not the case.)